olive/hardware/accelerator.py

```diff
@@ -171,199 +171,192 @@ def infer_single_device_from_execution_providers(execution_providers: List[str])
-    return system_config
-
-
-def fill_accelerators(system_config: "SystemConfig", system_supported_eps: List[str]):
-    """Fill the accelerators including device and execution providers in the system config.
-
-    * If the accelerators are not specified, fill the device and execution providers based on the installed ORT for
-      local/python system.
-    * If the device is specified but the execution providers are not, fill the execution providers based on the
-      installed ORT for local/python system.
-    * If the execution providers are specified but the device is not, fill the device based on the installed ORT.
-    """
-    if not system_config.config.accelerators:
-        # User does not specify the accelerators.
-        inferred_device = AcceleratorLookup.infer_single_device_from_execution_providers(system_supported_eps)
-        # here the pydantic validate_assignment will initialize the accelerator instances
-        system_config.config.accelerators = [{"device": inferred_device, "execution_providers": system_supported_eps}]
-        logger.info(
-            "There is no any accelerator specified. Inferred accelerators: %s",
-            system_config.config.accelerators,
-        )
-    else:
-        for accelerator in system_config.config.accelerators:
-            if not accelerator.device:
-                # User does not specify the device but providing the execution providers
-                assert accelerator.execution_providers, "The execution providers are not specified."
-                inferred_device = AcceleratorLookup.infer_single_device_from_execution_providers(
-                    accelerator.execution_providers
-                )
-                logger.info("the accelerator device is not specified. Inferred device: %s.", inferred_device)
-                accelerator.device = inferred_device
-            elif not accelerator.execution_providers:
-                # User specify the device but missing the execution providers
-                execution_providers = AcceleratorLookup.get_execution_providers_for_device_by_available_providers(
-                    accelerator.device.lower(), system_supported_eps
-                )
-                accelerator.execution_providers = execution_providers
-                filtered_eps = [ep for ep in system_supported_eps if ep not in execution_providers]
-                if filtered_eps:
-                    logger.warning(
-                        "The following execution providers are filtered: %s. "
-                        "Please raise issue in Olive site since it might be a bug. ",
-                        ",".join(filtered_eps),
+class AcceleratorNormalizer:
+    """Fill the device, execution providers and check the compatibility between the device and execution providers."""
+
+    def __init__(
+        self, system_config: "SystemConfig", skip_supported_eps_check: bool = True, is_ep_required: bool = True
```

Comment:
Is it possible to set `is_ep_required` as False by default? Since in olive/hardware level, it does not know the concept of EP which is a onnxruntime only config.

Reply:
make it default true or false doesn't matter since EP is part of hardware when running against ort.